test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -296,8 +296,71 @@ function ExecuteWebRequest
     return $result
 }
 
+[string] $verboseEncodingPrefix = 'Content encoding: '
+# This function calls Invoke-WebRequest with the given uri and
+# parses the verbose output to determine the encoding used for the content.
+function ExecuteRestMethod
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [string]
+        $Uri,
+
+        [switch] $UseBasicParsing
+    )
+    $result = [PSObject]@{Output = $null; Error = $null; Encoding = $null; Content = $null}
+    $verbosePreferenceSave = $VerbosePreference
+    $VerbosePreference = 'Continue'
+    try
+    {
+
+        $verboseFile = Join-Path $TestDrive -ChildPath ExecuteRestMethod.verbose.txt
+        $result.Output = Invoke-RestMethod -Uri $Uri -TimeoutSec 5 -UseBasicParsing:$UseBasicParsing.IsPresent -Verbose 4>$verboseFile
+        $result.Content = $result.Output
+
+        if (Test-Path -Path $verboseFile)
+        {
+            $result.Verbose = Get-Content -Path $verboseFile
+            foreach ($item in $result.Verbose)
+            {
+                $line = $item.Trim()
+                if ($line.StartsWith($verboseEncodingPrefix))
+                {
+                    $encodingName = $item.SubString($verboseEncodingPrefix.Length).Trim()
+                    $result.Encoding = [System.Text.Encoding]::GetEncoding($encodingName)
+                    break
+                }
+            }
+            if ($result.Encoding -eq $null)
+            {
+                throw "Encoding not found in verbose output. Lines: $($result).Verbose.Count Content:$($result.Verbose)"
+            }
+        }
+
+        if ($result.Verbose -eq $null)
+        {
+            throw "No verbose output was found"
+        }
+    }
+    catch
+    {
+        $result.Error = $_ | select-object * | Out-String
+    }
+    finally
+    {
+        $VerbosePreference = $verbosePreferenceSave
+        if (Test-Path -Path $verboseFile)
+        {
+            Remove-Item -Path $verboseFile -ErrorAction Ignore
```

Comment:
Usualy we use `SilentlyContinue` not `Ignore`.

Reply:
Gotcha, I'll change it.

Reply:
Closed.